src/middlewared/middlewared/plugins/apps/resources.py

```diff
@@ -151,11 +151,4 @@ async def get_hostpaths_datasets(self, app_name):
             if volume['source'].startswith(f'{IX_APPS_MOUNT_PATH}/') is False
         ]
 
-        mapping = {}
-        for host_path in host_paths:
-            try:
-                mapping[host_path] = await self.middleware.call('zfs.dataset.path_to_dataset', host_path)
-            except CallError:
-                mapping[host_path] = None
-
-        return mapping
+        return await self.middleware.call('zfs.dataset.paths_to_datasets', host_paths)
```

Comment:
Your change is better, but it still uses a child process for every apps host-path. That's bad. Please see the changes I introduced in https://github.com/truenas/middleware/pull/15272. Once those are merged,  you can call the `paths_to_datasets_impl` function directly. This does not use a child process and will be significantly faster/more efficient then what you're currently doing.